server/src/main/java/org/apache/druid/server/coordinator/rules/BroadcastDistributionRule.java

```diff
@@ -46,8 +46,9 @@ public CoordinatorStats run(DruidCoordinator coordinator, DruidCoordinatorRuntim
     } else {
       params.getDruidCluster().getAllServers().forEach(
           eachHolder -> {
-            if (colocatedDataSources.stream()
-                                    .anyMatch(source -> eachHolder.getServer().getDataSource(source) != null)) {
+            if (!eachHolder.isInMaintenance()
```

Comment:
Shouldn't this `!eachHolder.isInMaintenance()` condition be in an outer `if`, rather than in this if-else chain? As it is written, if a node in the maintenance mode, the coordinator will delete segments from it rather than leave the node alone.